VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java

```diff
@@ -54,27 +66,48 @@ public void call(Session session, SessionState state, Exception exception) {
         }
     };
 
+    @Nullable
+    private GetAccountTask getAccountTask;
+
+    ProfileModel profile;
+
+    ImageView imageView;
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         uiLifecycleHelper = IUiLifecycleHelper.Factory.getInstance(getActivity(), callback);
         uiLifecycleHelper.onCreate(savedInstanceState);
-    }
-
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-            Bundle savedInstanceState) {
-        super.onCreateView(inflater, container, savedInstanceState);
         Context context = getActivity().getBaseContext();
-
         socialPref = new PrefManager(context, PrefManager.Pref.FEATURES);
-
         pref = new PrefManager(context, PrefManager.Pref.LOGIN);
+        profile = pref.getCurrentUserProfile();
+        if (config.isUserProfilesEnabled() && profile != null && profile.username != null) {
+            getAccountTask = new GetAccountTask(getActivity(), profile.username) {
+                @Override
+                protected void onSuccess(Account account) throws Exception {
+                    final ProfileImage profileImage = account.getProfileImage();
+                    Glide.with(NavigationFragment.this)
+                            .load(profileImage.getImageUrlLarge())
```

Comment:
Despite the name, `large` is actually not very large, only 120px wide. So in fact, it is not really large enough, since we need 140px@2x, 210px@3x, and 280px@4x.

Unfortunately the next size up is 500px, but maybe we should just load that one.